storageservice/modules/blobs/utils.bal

```diff
@@ -143,28 +162,34 @@ isolated function createErrorFromXMLResponse(http:Response response) returns err
 
     match statusCode {
         http:STATUS_PRECONDITION_FAILED => {
-            return error PreconditionFailedError("Pre-conditions failed.", errorCode = errorCode, message = message);
+            return error PreconditionFailedError("Pre-conditions failed.", 
+            errorCode = errorCode, message = message, httpStatus = 412);
```

Comment:
Instead of hardcoding the statusCode, we can do like this
```suggestion
            errorCode = errorCode, message = message, httpStatus = statusCode);
```

Reply:
Purpose of sub-typing tests was to add a constant error code to each type. Due to a limitation, we cannot do it in error constructor, thus for now hard coded. I agree logic wise it is same, but we need to indicate, that this type is only for the errors with this status code only - so I think hard coding is better. https://github.com/ballerina-platform/ballerina-spec/issues/1219